pkg/collector/collector.go

```diff
@@ -637,8 +646,16 @@ func (cm *CollectorManager) FinalizeApplicationProfile(id *ContainerId) {
 	if _, ok := cm.containers[*id]; ok {
 		cm.containersMutex.Unlock()
 		// Patch the application profile to make it immutable with the final label
-		appProfileName := fmt.Sprintf("pod-%s", id.PodName)
-		_, err := cm.dynamicClient.Resource(AppProfileGvr).Namespace(id.Namespace).Patch(context.Background(),
+		var appProfileName string
+		var namespace string
+		if cm.config.StoreNamespace != "" {
+			appProfileName = fmt.Sprintf("pod-%s.%s", id.PodName, id.Namespace)
+			namespace = cm.config.StoreNamespace
+		} else {
+			appProfileName = fmt.Sprintf("pod-%s", id.PodName)
+			namespace = id.Namespace
+		}
+		_, err := cm.dynamicClient.Resource(AppProfileGvr).Namespace(namespace).Patch(context.Background(),
```

Comment:
same as above, a reusable function would be great

Reply:
done